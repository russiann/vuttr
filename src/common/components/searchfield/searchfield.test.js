import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Searchfield from './searchfield.component';

afterEach(cleanup);

it('snapshot', () => {
  const {asFragment} = render(<Searchfield />);
  expect(asFragment()).toMatchSnapshot();
});

it('should render default value', () => {
  const text = 'Hello World!';
  const {getByTestId} = render(<Searchfield defaultValue={text} />);
  const input = getByTestId('input');
  expect(input.value).toBe(text);
});

it('should render typed text', () => {
  const text = 'Hello World!';
  const {getByTestId} = render(<Searchfield />);
  const input = getByTestId('input');
  fireEvent.change(input, {target: {value: text}});
  expect(input.value).toBe(text);
});

it('should fire onSearch with value when button clicked', () => {
  const onSearchSpy = jest.fn();
  const value = 'lorem ipsum dolor';
  const {getByTestId} = render(
    <Searchfield onSearch={onSearchSpy} value={value} onChange={jest.fn} />
  );
  const button = getByTestId('button');
  fireEvent.click(button);
  expect(onSearchSpy).toHaveBeenCalledWith(value);
});

it('should fire onSearch when keydown Enter', async () => {
  const onSearchSpy = jest.fn();
  const value = 'lorem ipsum dolor';

  const {getByTestId} = render(
    <Searchfield onSearch={onSearchSpy} value={value} onChange={jest.fn} />
  );

  const input = getByTestId('input');

  fireEvent.keyDown(input, {
    key: 'Enter',
    code: 13,
    keyCode: 13
  });

  expect(onSearchSpy).toHaveBeenCalledWith(value);
});
