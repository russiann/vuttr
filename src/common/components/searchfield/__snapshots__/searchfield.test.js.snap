// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  .emotion-6 {
  border-radius: 30px;
  border: 1px solid #bdbdbd;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  overflow: hidden;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #bdbdbd;
  padding-left: 12px;
}

.emotion-6 .emotion-5 {
  margin-left: auto;
}

.emotion-0 {
  width: 17px;
  height: 17px;
}

.emotion-2 {
  font-family: 'Ubuntu',sans-serif;
  width: 100%;
  outline: none;
  padding: 10px;
  border: none;
  min-width: 200px;
  font-size: 12px;
}

.emotion-4 {
  font-family: 'Ubuntu',sans-serif;
  font-size: 12px;
  color: #fff;
  background-color: #ff6007;
  text-align: center;
  text-transform: uppercase;
  padding: 10px 50px;
  border: none;
  border-radius: 30px;
  outline: none;
  cursor: pointer;
}

.emotion-4:hover {
  opacity: 0.9;
}

<div
    class="emotion-6 emotion-7"
  >
    <div
      class="emotion-0 emotion-1"
    >
      <svg
        fill="none"
        height="17"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
    </div>
    <input
      class="emotion-2 emotion-3"
      data-testid="input"
      value=""
    />
    <button
      class="emotion-4 emotion-5"
      data-testid="button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
